tests/cases/compiler/asyncFunctionContextuallyTypedReturns.ts

```diff
@@ -0,0 +1,14 @@
+// @target: es6
+// @strict: true
+declare function f(cb: (v: boolean) => [0] | PromiseLike<[0]>): void;
+f(v => v ? [0] : Promise.reject()); // ok
+f(async v => v ? [0] : Promise.reject()); // error

```

Comment:
(not an error anymore)

```suggestion
f(v => v ? [0] : Promise.reject());
f(async v => v ? [0] : Promise.reject());
```